### ISSUE TITLE
OneHot Encoding - Pytorch

### ISSUE THREAD

**Issue by vmtmxmf5** [vmtmxmf5/Pytorch-#3]
```sage
y_oh = torch.zeros_like(y_pred)
y_oh.scatter_(1, y_train.unsqueeze(1), 1)
```
scatter_ : in-place
scatter_(dim, idx, 채울 숫자)

**Comment by vmtmxmf5** [comment]
## Cross Entropy 정리

| tf | torch |
|--|--|
| tf.losses.sparse_categorcial_crossentropy | 직접 정의 <br> (y_oh * -torch.log(y_hat)).sum(dim=1).mean() |
| tf.losses.categorical_crossentropy | F.Cross_entropy(z, y_train) <br> softmax까지 한 번에 합친 함수 |

```sage
for epoch in range(2000):
    # OH
    y_hat = F.softmax(X_train.matmul(W) + b, dim=1)
    cost = (y_oh * -torch.log(y_hat)).sum(dim=1).mean()
    # sum(dim=1) : 원핫과 계산해서 0이 된 열들을 하나로 모아주는 작업
    
    # ~OH
    z = X_train.matmul(W) + b
    cost = F.cross_entropy(z, y_train)
    
    optm.zero_grad()
    cost.backward()
    optm.step()
    
    if epoch % 100 == 0:
        print(cost.item())
```
